r/src/compute.cpp

```diff
@@ -123,19 +123,19 @@ arrow::Datum as_cpp<arrow::Datum>(SEXP x) {
 SEXP from_datum(arrow::Datum datum) {
```

Comment:
Someday this should probably be
```c++
SEXP as_sexp(arrow::Datum datum)
```
so that we don't need to wrap datums in `from_datum`

Reply:
Would that still do the dispatch internally before reaching the R side or would there be an R6 class for `arrow::Datum`. Probably the latter.

Reply:
I don't think an R6 class for `Datum` has value, since SEXP already fills the role of a discriminated union of the members of Datum.

With the addition of `R6` I amend my earlier comment: `from_datum` probably be replaced by `R6::R6(Datum)`